lib/amazonka/src/Amazonka/Env.hs

```diff
@@ -42,20 +41,36 @@ module Amazonka.Env
   )
 where
 
-import Amazonka.Auth
 import Amazonka.Lens ((.~), (?~))
 import Amazonka.Prelude
 import Amazonka.Types
 import Control.Lens (Lens)
 import qualified Data.Function as Function
 import Data.Monoid (Dual (..), Endo (..))
+import qualified Data.Text as Text
 import qualified Network.HTTP.Client as Client
 import qualified Network.HTTP.Conduit as Client.Conduit
+import System.Environment as Environment
 
 type Env = Env' Identity
 
 type EnvNoAuth = Env' Proxy
 
+-- | The environment containing the parameters required to make AWS requests.
+--
+-- This type tracks whether or not we have credentials at the type
+-- level, to avoid "presigning" requests when we lack auth
+-- information.
+data Env' withAuth = Env
+  { _envRegion :: Region,
+    _envLogger :: Logger,
+    _envRetryCheck :: Int -> Client.HttpException -> Bool,
+    _envOverride :: Dual (Endo Service),
+    _envManager :: Client.Manager,
+    _envAuth :: withAuth Auth
+  }
+  deriving stock (Generic)
```

Comment:
If `Generic` is used for `generic-lens` or `optics` support - the `_` prefixes at the least should probably also be removed.

Reply:
If they should be removed, they should also be removed in e.g. `AuthEnv(_authAccessKeyId)` from Amazonka/Types.hs, no?

Reply:
Yes, it would be nice to remove them fully from the library - although perhaps this would be better as a separate PR.

Reply:
Agreed. I'll do the ones on `Env'` now, and defer a general record cleanup to https://github.com/brendanhay/amazonka/issues/752.